build-packages.sh

```diff
@@ -87,7 +95,7 @@ while :; do
         ;;
         -BuildArch=*)
         unprocessedBuildArgs="$unprocessedBuildArgs $1"
-        __Arch=$(echo $1| cut -d'=' -f 2)
+        __HostArch=$(echo $1| cut -d'=' -f 2)
```

Comment:
I'm not sure `HostArch` is the right term here, is it? If I'm building for `arm`, but on an `x64` machine, what is `HostArch`?  In my mind, `HostArch` is the architecture of the machine we are building on (i.e. the "host").  I think it would make the most sense to align this name with the external option name: `BuildArch`.

In core-setup, we have been moving to the following terms to try to eliminate any confusion:
* What is the RID of the current machine?  $(HostMachineRid)
* What is the RID of the toolset we are using to build?  core-setup doesn't have this scenario yet, but corefx calls this $(ToolRuntimeRID)
* What is the RID that the current build is producing outputs for?  $(OutputRid)